machine/v1beta1/types_azureprovider.go

```diff
@@ -183,6 +183,19 @@ type OSDisk struct {
 	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
 	// DiskSizeGB is the size in GB to assign to the data disk.
 	DiskSizeGB int32 `json:"diskSizeGB"`
+	// DiffDiskSettings describe ephemeral disk settings for the os disk.
+	// +optional
+	DiffDiskSettings *DiffDiskSettings `json:"diffDiskSettings,omitempty"`
```

Comment:
Does this need to be a pointer? Is there a benefit to it being a pointer?

As a user this means there is a difference between not setting the option and setting:
```
diffDiskSettings: {}
```
Would that make a difference in the code, and why?

Reply:
I believe this comes from here https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/0ac1befc0262b22b35673fcf2473ffd69d8ec9cb/api/v1beta1/types.go#L458

Reply:
@alexander-demichev is right. it's done to make our API the same as in the upstream.

Reply:
i thought we were having a similar discussion about pointers in APIs with the Alibaba refactor stuff?

do we have a reference to a guideline about not using pointers in kube APIs?

i generally agree with you about keeping the API clean, i'm just curious about the official guidance on this since it seems to have come up several times recently.

Reply:
I asked in slack and got confirmation https://coreos.slack.com/archives/CE4L0F143/p1637083434049900, it is expected we would not use a pointer to the struct here

Reply:
Updated. Now DiffDiskSettings is a struct.